packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -36,18 +36,42 @@ enum _ScaffoldSlot {
   statusBar,
 }
 
+/// Geometry information for scaffold components.
+///
+/// To get a [ValueNotifier] for the scaffold gemoetry call
+/// [Scaffold.geometryOf].
+@immutable
+class ScaffoldGeometry {
+  /// The rectangle in which the scaffold is laying out
+  /// [Scaffold.bottomNavigationBar].
+  /// Null when there is no bottom navigation bar..
+  final Rect bottomNavigationBar;
+
+  /// The rectangle in which the scaffold is laying out
+  /// [Scaffold.floatingActionButton].
+  /// Null when there is no bottom navigation bar..
+  final Rect floatingActionButton;
```

Comment:
similarly here i'd probably express this in terms of the offset from some reference point to the center of the fab, and a double giving the current fractional size of the fab. That would give more flexibility, e.g. in case the shape of the fab is not well described by a Rect (e.g. it's a T-shape, or a diamond).

Reply:
This will be a fraction of what? of some fixed fab size? if we want to reduce this to a single double maybe we can use the size of one of the edges, or the diagonal, though I'm not sure this the best thing to do here...

Your point for reducing the temptation to use the bottomNavigationBarRect makes sense when thinking of the BAB use case, where we want to at least look at the BAB's size.

For the FAB dimensions though, there is no such temptation, in the BAB use case all you know about the FAB comes from ScaffoldGeometry.

I would leave it a Rect as it contains the full information of where the scaffold is laying out the FAB in a simple form. Also you could think of a case where the width/height proportions are changing, and a fractional size is not enough.